phoenix-core/src/main/java/org/apache/phoenix/parse/ColumnParseNode.java

```diff
@@ -40,6 +41,13 @@ public ColumnParseNode(TableName tableName, String name, String alias) {
         this.tableName = tableName;
         fullName = tableName == null ? getName() : tableName.toString() + QueryConstants.NAME_SEPARATOR + getName();
     }
+    
+    public ColumnParseNode(TableName tableName, String name, String alias, boolean caseSensitive) {
```

Comment:
We should be able to store the expressionStr for a functional index in a case sensitive manner. You'll get this for free when you parse and compile the functional expression. If a referenced column is not surrounded with double quotes, then it'll be upper cased, otherwise it won't. Then the equality check will always be a straight equals check.